### DISCUSSION TITLE
vpn support , like zerotier.

### DISCUSSION THREAD

**Discussion by gemron** [rustdesk/rustdesk#1136] [General]
The control terminal and the controlled terminal can generate private IP addresses and can access each other. zerotier uses p2p technology, it may be implemented by integrating its development library, or it can be implemented using rust again.

**Comment by rustdesk** [comment]
Technically, yes

**Reply by gemron** [reply]
This function allows developers and deployment personnel to remotely access the services published by the controlled computer, which is convenient for remote deployment and debugging. For ordinary users, web 3.0 decentralized applications can be provided, of course, this is just an expectation.

**Reply by yuluo-zy** [reply]
> Technically, yes

Has the company considered funding this project? I believe I can complete the built-in VPN module. However, this module will require significant modifications and necessary discussions.

**Reply by Thaina** [reply]
If this is not too much of a hassle I wish rustdesk would give ipaddress we could use for ssh since the connection already established

**Reply by lurenjia1213** [reply]
use easytier
